### PR TITLE
Add circle ci config for front only project

### PR DESCRIPTION
**Pull request by georgesbiaux**

### CODE REVIEW

File: generators/react/templates/.circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - checkout
+      - run:
+          name: Install yarn
+          command: 'sudo npm install -g yarn@latest'
+      - restore_cache:
+          key: front-dependency-cache-{{ checksum "~/<%= appName %>/package.json" }}
+      - run:
+          name: Install front dependencies
+          command: cd ~/<%= appName %> && yarn install
```
Comment (ghamaide) [inline generators/react/templates/.circleci/config.yml]: `yarn install` or `yarn`

File: generators/react/templates/.circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  test_frontend:
+    working_directory: ~/<%= appName %>
+    docker:
+      - image: circleci/node:8.4.0
+    steps:
+      - checkout
+      - run:
+          name: Install yarn
+          command: 'sudo npm install -g yarn@latest'
```
Comment (ghamaide) [inline generators/react/templates/.circleci/config.yml]: yarn@latest - it might be breaking some day no ?

**Comment by Remy-Luciani** [comment]
Hey @georgesbiaux ,

We had a deployment failure on MyAuto, because the front app production build was broken. The staging env was down because of the symlink done even without a good deploy, we fixed the issue etc. All of this is not related. The real problem is that a breaking commit has been merged to the develop branch.

In order to fail faster, I have added the build in production mode of the app into the CI runs. If the prod build is broken, CI fails.

So, I Suggest adding a task in your template:

```
- run:
  name: Build production app
  command: cd ~/<%= appName %> && yarn build
```

What do you think?

cc @ghamaide

**Comment by georgesbiaux** [comment]
@Remy-Luciani I like the idea, I will implement this :)

**Comment by ghamaide** [comment]
@Remy-Luciani Great idea, we had the same issue on my project last week.

File: generators/react/templates/.circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      - restore_cache:
+          key: front-dependency-cache-{{ checksum "~/<%= appName %>/package.json" }}
```
Comment (williamdclt) [inline generators/react/templates/.circleci/config.yml]: No need for the `~/<%= appName %>/` part, the `working_directory` is set above

File: generators/react/templates/.circleci/config.yml (hunk 1 above)
Comment (williamdclt) [inline generators/react/templates/.circleci/config.yml]: Same, no need to `cd`

File: generators/react/templates/.circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      - run:
+          name: Install front dependencies
+          command: cd ~/<%= appName %> && yarn install
+      - save_cache:
+          key: front-dependency-cache-{{ checksum "~/<%= appName %>/package.json" }}
+          paths:
+            - node_modules
+      - run:
+          name: Test
+          command: cd  ~/<%= appName %> && yarn test:coverage
```
Comment (williamdclt) [inline generators/react/templates/.circleci/config.yml]: same

File: generators/react/templates/.circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      - store_artifacts:
+          path: ~/<%= appName %>/coverage
```
Comment (williamdclt) [inline generators/react/templates/.circleci/config.yml]: same

File: generators/react/templates/.circleci/config.yml (hunk 2 above)
Comment (williamdclt) [inline generators/react/templates/.circleci/config.yml]: Not necessary, yarn is already installed
https://circleci.com/docs/2.0/yarn/#using-yarn-in-circleci

File: generators/react/templates/.circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+version: 2
+jobs:
+  test_frontend:
+    working_directory: ~/<%= appName %>
+    docker:
+      - image: circleci/node:8.4.0
```
Comment (williamdclt) [inline generators/react/templates/.circleci/config.yml]: Missing a
```
    environment:
      NODE_ENV: test
```
somewhere around here

**Comment by georgesbiaux** [comment]
@williamdclt @ghamaide @Remy-Luciani I took your requests into account. Please review :)

**Pull request merged**